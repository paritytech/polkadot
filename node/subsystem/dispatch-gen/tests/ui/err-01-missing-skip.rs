#![allow(dead_code)]

use polkadot_procmacro_subsystem_dispatch_gen::subsystem_dispatch_gen;

/// The event type in question.
#[derive(Clone, Copy)]
enum Event {
	Smth,
	Else,
}

impl Event {
	fn focus(&self) -> std::result::Result<Inner, ()> {
		unimplemented!("foo")
	}
}

/// This should have a `From<Event>` impl but does not.
#[derive(Clone)]
enum Inner {
	Foo,
	Bar(Event),
}

#[subsystem_dispatch_gen(Event)]
#[derive(Clone)]
enum AllMessages {
	/// Foo
	Vvvvvv(Inner),

    /// Missing a `#[skip]` annotation
    Uuuuu,
}

fn main() {
    let _x = AllMessages::dispatch_iter(Event::Else);
}
