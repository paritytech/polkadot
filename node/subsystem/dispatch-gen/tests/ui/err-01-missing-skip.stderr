error: Must be annotated with #[skip].
  --> $DIR/err-01-missing-skip.rs:32:5
   |
32 |     Uuuuu,
   |     ^^^^^

error[E0599]: no variant or associated item named `dispatch_iter` found for enum `AllMessages` in the current scope
  --> $DIR/err-01-missing-skip.rs:36:27
   |
27 | enum AllMessages {
   | ---------------- variant or associated item `dispatch_iter` not found here
...
36 |     let _x = AllMessages::dispatch_iter(Event::Else);
   |                           ^^^^^^^^^^^^^ variant or associated item not found in `AllMessages`
